src/Materials/PBR/pbrBaseMaterial.ts

```diff
@@ -922,12 +875,12 @@ export abstract class PBRBaseMaterial extends PushMaterial {
     /**
      * Defines the clear coat layer parameters for the material.
      */
-    public readonly clearCoat = new PBRClearCoatConfiguration(this._markAllSubMeshesAsTexturesDirty.bind(this));
+    public readonly clearCoat: PBRClearCoatConfiguration;

```

Comment:
shouldn t those be nullable if uninitialized ? I am afraid of typings not highlighting the possibility it is not available but this would be a breaking change so it should always get imported for the current one.

Reply:
With the static plugin registration these are always initialized.

Reply:
Not always when using ES6 if the corresponding material plugin files are not imported.

But I don't see a way to do it otherwise, it's up to the user to not use the property (`clearCoat`, `sheen`, `subSurface`, etc) if they have not imported the corresponding class.

Reply:
Ah, I see. But this means that the current code will break all existing apps that import modules directly for tree shaking, right? In the current code, if the `PBR/index.ts` file isn't imported, all these plugins won't be. Perhaps `pbrBaseMaterial` should import them directly?

Reply:
Yes, to avoid a breaking change I think `PBRBaseMaterial` should import them all, but it means it won't be possible to tree-shake finely with PBR (for eg, only import sheen config if you only use sheen).

As 5.0 is quite a breaking change release anyway, maybe we can document this one and let people know they should import the right config classes depending on the PBR features they are using. It's an anoyance but in return you save some space in download size (which is the first reason to use ES6 I think). @sebavan / @deltakosh ?

Reply:
yup let s check with @RaananW how could we solve this with the LTS system ?

pbrBaseMaterial in one could have all of them without where the other one could be a nullable type ?